lib/ansible/modules/cloud/amazon/aws_inspector_template.py

```diff
@@ -0,0 +1,581 @@
+        botocore.exceptions.ClientError,
+    ) as e:
+        module.fail_json_aws(e, msg="trying to delete sub from target")
+
+    return changed
+
+
+def update_subs(client, module, arn, subs):
+    changed = False
+    existing_subs = _retrieve_subs(client, module, arn)
+
+    to_add = [sub for sub in subs if sub not in existing_subs]
+    if to_add:
+        if add_subs(client, module, arn, to_add):
+            changed = True
+
+    to_delete = [sub for sub in existing_subs if sub not in subs]
+    if to_delete:
+        if delete_subs(client, module, arn, to_delete):
+            changed = True
+
+    return changed
+
+
+def get_target_arn(client, module, target):
+    try:
+        return list_assessment_targets_with_backoff(
+            client,
+            filter={'assessmentTargetNamePattern': target},
+        ).get('assessmentTargetArns')[0]
+    except (
+        botocore.exceptions.BotoCoreError,
+        botocore.exceptions.ClientError,
+    ) as e:
+        module.fail_json_aws(e, msg="trying to retrieve target arn")
+    except IndexError:
+        module.fail_json(msg='unknown target: %s' % target)
+
+
+def main():
+    argument_spec = dict(
+        attributes=dict(type='dict', default={}),
+        duration=dict(
+            type='int',
+            choices=[15, 60, 480, 720, 1440],
+            default=60,
+        ),
+        name=dict(required=True),
+        rules=dict(type='list', default=[]),
+        subscriptions=dict(type='list', default=[]),
+        state=dict(choices=['absent', 'present'], default='present'),
+        tags=dict(type='dict', default={}),
+        target=dict(),
+    )
+
+    required_if = [['state', 'present', ['rules', 'target']]]
+
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        required_if=required_if,
+        supports_check_mode=False,
+    )
+
+    attributes = module.params.get('attributes')
+    duration = module.params.get('duration')
+    name = module.params.get('name')
+    rules = module.params.get('rules')
+    subscriptions = _check_subs(module, module.params.get('subscriptions'))
+    state = module.params.get('state').lower()
+    tags = module.params.get('tags')
+    target = module.params.get('target')
+
+    retry_decorator = AWSRetry.exponential_backoff(
+        retries=5,
+        delay=5,
+        backoff=2.0,
+    )
+    client = module.client('inspector', retry_decorator=retry_decorator)
+    arn = get_template_arn(client, module, name)
+    if state == 'present' and arn:
+        changed = False
+
+        if update_tags(client, module, arn, tags):
+            changed = True
+
+        if update_subs(client, module, arn, subscriptions):
+            changed = True
+
+        module.exit_json(
+            changed=changed,
+            **get_template(client, module, arn)
+        )
+    elif state == 'present' and not arn:
+        try:
+            arn = create_assessment_template_with_backoff(
```

Comment:
So down here, instead of needing a separate `_with_backoff()` function, you could write this code:

```
arn = client.create_assessment_template(
    aws_retry=True
    assessmentTargetArn=get_target_arn(client, module, target),
    assessmentTemplateName=name,
    durationInSeconds=duration * 60,
    rulesPackageArns=get_rules_arns(client, module, rules),
    userAttributesForFindings=ansible_dict_to_boto3_tag_list(
        attributes, 'key', 'value'
    )
)
```